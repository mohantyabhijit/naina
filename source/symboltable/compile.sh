#!/bin/bash

###############################################################
# This is a script that compiles the C files in this directory#
###############################################################

declare -a arr=("create" "insert" "symboltable" "search")

for i in "${arr[@]}"
do
	gcc -c -o "$i".o "$i".c
done
