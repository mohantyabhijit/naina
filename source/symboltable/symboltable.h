#ifndef SYMBOLTABLE_H
#define SYMBOLTABLE_H

#include<stdio.h>
#include<string.h>
#include<unistd.h>
#include<stdlib.h>

int symboltable(char *fun, void *psym, void *padd);
int create(char *ds);
int search(char *sym);
int insert(void *sym, void *add);

#ifndef TRUE
#define TRUE 0
#endif

#ifndef FALSE
#define FALSE 1
#endif

#ifndef MAXSIZE
#define MAXSIZE 50
#endif

#define TABLECREATED 11
#define SYMWRITE 12
#define FOUND 13

#define	TABLEERR 101
#define SYMERR 102
#define TRUENOTEXIST 103
#define NOTFOUND 104

#endif
