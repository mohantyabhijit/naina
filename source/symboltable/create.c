#include"symboltable.h"

int create(char *ds)
{
	return FALSE;
}
