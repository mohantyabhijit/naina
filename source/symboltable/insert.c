#include"symboltable.h"

int insert(void *sym, void *add)
{
	return FALSE;
}
