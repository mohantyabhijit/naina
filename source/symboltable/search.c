#include"symboltable.h"

int search(char *sym)
{
	return FALSE;
}
