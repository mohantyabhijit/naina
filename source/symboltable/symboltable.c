#include"symboltable.h"

int symboltable(char *fun, void *psym, void *padd)
{
	return FALSE;
}
