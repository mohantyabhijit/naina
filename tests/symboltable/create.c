// This is a basic template of a test file. Modify it according to the need of the function
#include"../../source/symboltable/symboltable.h"

#define MAXCASE 8


int main(int argc, char *argv[])
{
	char *tmpIn=NULL, tmpOut[MAXSIZE];
	int loop = 0, val =0;
	size_t length=0;
	FILE *pfin, *pfout, *pftmp;
	pfin = fopen("./create.in", "r");
	pfout = fopen("./create.out", "r");
	pftmp = fopen("./create.tmp", "w+");
	memset(tmpOut, '\0', MAXSIZE);
	for (loop=0;loop<MAXCASE;loop++)
	{
		getline(&tmpIn, &length, pfin);
		val = create(tmpIn);
		switch (val)
		{
			case TABLECREATED:
			{
				strncpy(tmpOut, "TABLECREATED", 15);
				break;
			}
			case TABLEERR:
			{
				strncpy(tmpOut, "TABLEERR", 15);
				break;
			}
			default:
			{
				strncpy(tmpOut, "FALSE", 8);
				break;
			}
		}
		strncat(tmpOut, "\n", 2);
		fputs(tmpOut, pftmp);
	}
	fclose(pfin);
	fclose(pfout);
	fclose(pftmp);
	return 0;
}
