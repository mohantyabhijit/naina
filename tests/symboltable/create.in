LARRAY	
LLIST	
BST
HTABLE
HTABL
STACK
LARRAY
BST
