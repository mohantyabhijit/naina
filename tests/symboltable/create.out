TABLECREATED
TABLECREATED
TABLECREATED
TABLECREATED
TABLEERR
TABLEERR
TABLEERR
TABLEERR

